tests/queries/0_stateless/03629_starts_endswith_caseinsensitive.sql

```diff
@@ -39,3 +39,46 @@ SELECT endsWithUTF8CaseInsensitive('中国', '国'), endsWithUTF8CaseInsensitive
 SELECT '-- Test invalid UTF8';
 SELECT startsWithCaseInsensitive('中国', '\xe4'), startsWithUTF8CaseInsensitive('中国', '\xe4');
 SELECT endsWithCaseInsensitive('中国', '\xbd'), endsWithUTF8CaseInsensitive('中国', '\xbd');
+
+DROP TABLE IF EXISTS tab;
+
+CREATE TABLE tab(S1 String, S2 String, S3 FixedString(4)) ENGINE=Memory;
+INSERT INTO tab values ('1a', 'a', 'AbA'), ('22', 'A', 'ab'), ('中国', '中', '国');
+
+SELECT '-- Test constant needle with haystack - prefix';
+SELECT COUNT() FROM tab WHERE startsWithCaseInsensitive(S1, '1');
+SELECT COUNT() FROM tab WHERE startsWithCaseInsensitive(S2, '中');
+SELECT COUNT() FROM tab WHERE startsWithCaseInsensitive(S3, '国');
+
+SELECT COUNT() FROM tab WHERE startsWithUTF8CaseInsensitive(S1, '1');
+SELECT COUNT() FROM tab WHERE startsWithUTF8CaseInsensitive(S2, '中');
+-- startsWithCaseUTF8CaseInsensitive does not support FixedString
+
+SELECT '-- Test constant needle with haystack - suffix';
+SELECT COUNT() FROM tab WHERE endsWithCaseInsensitive(S1, '2');
+SELECT COUNT() FROM tab WHERE endsWithCaseInsensitive(S2, '中');
+SELECT COUNT() FROM tab WHERE endsWithCaseInsensitive(S3, '国\0');
+
+SELECT COUNT() FROM tab WHERE endsWithUTF8CaseInsensitive(S1, '2');
+SELECT COUNT() FROM tab WHERE endsWithUTF8CaseInsensitive(S2, '中');
+-- endsWithCaseUTF8CaseInsensitive does not support FixedString
+
+SELECT '-- Test column needle with haystack - prefix';
+SELECT COUNT() FROM tab WHERE startsWithCaseInsensitive(S1, S1);
+SELECT COUNT() FROM tab WHERE startsWithCaseInsensitive(S1, S2);
+SELECT COUNT() FROM tab WHERE startsWithCaseInsensitive(S2, S3);
+
+SELECT COUNT() FROM tab WHERE startsWithUTF8CaseInsensitive(S1, S1);
+SELECT COUNT() FROM tab WHERE startsWithUTF8CaseInsensitive(S1, S2);
+-- endsWithCaseUTF8CaseInsensitive does not support FixedString
+
+SELECT '-- Test column needle with haystack - suffix';
+SELECT COUNT() FROM tab WHERE endsWithCaseInsensitive(S1, S1);
+SELECT COUNT() FROM tab WHERE endsWithCaseInsensitive(S1, S2);
+SELECT COUNT() FROM tab WHERE endsWithCaseInsensitive(S2, S3);
+
+SELECT COUNT() FROM tab WHERE endsWithUTF8CaseInsensitive(S1, S1);
```

Comment:
@vdimir, pay attention at the consistency of the names:

positionCaseInsensitiveUTF8
endsWithUTF8CaseInsensitive

Rewrite accordingly.